internal/pkg/api/repository/result/mongo_test.go

```diff
@@ -0,0 +1,228 @@
+		assert.NoError(err)
+		assert.Empty(executions)
+	})
+
+	t.Run("getting totals with script name that doesn't exist should return 0 results", func(t *testing.T) {
+		totals, err := repository.GetExecutionTotals(context.Background(), NewExecutionsFilter().WithScriptName("noneExisting"))
+
+		assert.NoError(err)
+		assert.Equal(int32(0), totals.Results)
+		assert.Equal(int32(0), totals.Failed)
+		assert.Equal(int32(0), totals.Passed)
+		assert.Equal(int32(0), totals.Queued)
+		assert.Equal(int32(0), totals.Pending)
+	})
+
+	t.Run("filter with ccombined filter should return corresponding results", func(t *testing.T) {
+		filter := NewExecutionsFilter().
+			WithStatus(testkube.SUCCESS_ExecutionStatus).
+			WithStartDate(twoDaysAgo).
+			WithEndDate(oneDayAgo).
+			WithScriptName(defaultName)
+		executions, err := repository.GetExecutions(context.Background(), filter)
+		assert.NoError(err)
+		assert.Len(executions, 2)
+	})
+
+	t.Run("getting totals with ccombined filter should return corresponding results", func(t *testing.T) {
+		filter := NewExecutionsFilter().
+			WithStatus(testkube.SUCCESS_ExecutionStatus).
+			WithStartDate(twoDaysAgo).
+			WithEndDate(oneDayAgo).
+			WithScriptName(defaultName)
+		totals, err := repository.GetExecutionTotals(context.Background(), filter)
+
+		assert.NoError(err)
+		assert.Equal(int32(2), totals.Results)
+		assert.Equal(int32(0), totals.Failed)
+		assert.Equal(int32(2), totals.Passed)
+		assert.Equal(int32(0), totals.Queued)
+		assert.Equal(int32(0), totals.Pending)
+	})
+
+	name := "someDifferentName"
+	err = repository.insertExecutionResult(name, testkube.PENDING_ExecutionStatus, twoDaysAgo)
+	assert.NoError(err)
+
+	t.Run("filter with script name should return result only for that script name", func(t *testing.T) {
+
+		executions, err := repository.GetExecutions(context.Background(), NewExecutionsFilter().WithScriptName(name))
+		assert.NoError(err)
+		assert.Len(executions, 1)
+		assert.Equal(executions[0].ScriptName, name)
+	})
+
+	t.Run("getting totals with script name should return result only for that script name", func(t *testing.T) {
+		totals, err := repository.GetExecutionTotals(context.Background(), NewExecutionsFilter().WithScriptName(name))
+
+		assert.NoError(err)
+		assert.Equal(int32(1), totals.Results)
+		assert.Equal(int32(0), totals.Failed)
+		assert.Equal(int32(0), totals.Passed)
+		assert.Equal(int32(0), totals.Queued)
+		assert.Equal(int32(1), totals.Pending)
+	})
+
+}
+
+func getRepository() (*MongoRepository, error) {
+	db, err := storage.GetMongoDataBase(mongoDns, mongoDbName)
+	repository := NewMongoRespository(db)
+	return repository, err
+}
+
+func (repository *MongoRepository) insertExecutionResult(scriptName string, execStatus testkube.ExecutionStatus, startTime time.Time) error {
+	return repository.Insert(context.Background(),
+		testkube.Execution{
+			Id:              "dummyID",
```

Comment:
We should create unique index for Id - so this will fail in future

Reply:
Done, please check the latest commit.